### PR TITLE
feat(input, input-number, input-text): improve keyboard navigation by preventing the clear button from being focusable

### PR DESCRIPTION
**Pull request by Apahadi73**
**Related Issue:** #2381

## Summary

This PR removes ability to perform keyboard navigation on input clear button as discussed in the issue: [Question: Should input clearable button be keyboard navigable? #2381](https://github.com/Esri/calcite-components/issues/2381)

### CODE REVIEW

**Review by jcfranco — CHANGES_REQUESTED** [review]
@Apahadi73 Thanks for helping out with this one! I have a few comments:

* `calcite-input-text` needs to be updated as well
* The E2E tests suites will need to be updated to cover the changes made
* The scope of this commit affects more than `input`, so it needs to reflect this (I can tackle this)

Once the above are done, I will proceed to 'adopt' the PR so it goes through all checks and can then get merged.

Let us know if you have any questions.

**Comment by Apahadi73** [comment]
> @Apahadi73 Thanks for helping out with this one! I have a few comments:
> 
> * `calcite-input-text` needs to be updated as well
> * The E2E tests suites will need to be updated to cover the changes made
> * The scope of this commit affects more than `input`, so it needs to reflect this (I can tackle this)
> 
> Once the above are done, I will proceed to 'adopt' the PR so it goes through all checks and can then get merged.
> 
> Let us know if you have any questions.



Thanks, @jcfranco for all of the suggestions. I have made all of the suggested changes. This PR is ready for final review now.

File: src/components/pick-list/shared-list-tests.ts [hunk 1]
```diff
@@ -186,7 +186,7 @@ export function keyboardNavigation(listType: ListType): void {
       await page.keyboard.press("Tab");
       await page.keyboard.press("Tab");
 
-      expect(await getFocusedItemValue(page)).toEqual("one");
```
Comment (jcfranco) [inline src/components/pick-list/shared-list-tests.ts:189]: This test is unrelated. Can you revert it and update the corresponding input E2E tests:

* https://github.com/Esri/calcite-components/blob/master/src/components/input-number/input-number.e2e.ts
* https://github.com/Esri/calcite-components/blob/master/src/components/input-text/input-text.e2e.ts
* https://github.com/Esri/calcite-components/blob/master/src/components/input/input.e2e.ts

?

Comment (jcfranco) [reply src/components/pick-list/shared-list-tests.ts:189]: Apologies, this is related, but needs to be updated differently. The double tab key presses were meant to move focus from the input, to the clear button and then to the next item. Since this makes the clear button not focusable, that second tab press should no longer be needed (should now tab from input to item). The assertion needs to stay the same.

Comment (Apahadi73) [reply src/components/pick-list/shared-list-tests.ts:189]: Hello @jcfranco, I have applied your suggestion. ☺️

Comment (jcfranco) [reply src/components/pick-list/shared-list-tests.ts:189]: Awesome! Thanks for making the changes. 

I think we'll be able to merge this w/o adopting. 🎉  I scheduled another E2E test run since it looks like it failed on an unrelated test.

**Review by jcfranco — APPROVED** [review]
💥

**Comment by jcfranco** [comment]
@Apahadi73 Thanks again for working on this! 🏆
